spec/support/importer.rb

```diff
@@ -23,3 +17,20 @@
     importer.import
   end
 end
+
+shared_examples 'a versionable resource' do
+  before { described_class.name.gsub(/Data$/, '').constantize.recreate_index }
```

Comment:
In fact, I'll change it myself... the rest of the PR looks fine, but I'll do some thorough testing before merging.